From 516db20f24938aa5482334d2726d70e9935ed32b Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Thu, 22 Oct 2015 22:22:58 -0500
Subject: [PATCH 1142/1691] MLK-11751: mtd: gpmi: add empty sentinel entry at
 the end of of_device_id table

commit 516db20f24938aa5482334d2726d70e9935ed32b from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

add an empty sentinel entry to avoid the struct of_device_id is not
terminated with a NULL entry issue.

Signed-off-by: Han Xu <b45815@freescale.com>
(cherry picked from commit 2b1ce5ec442dde8801b6b2d059d22f5dce7c9c76)
---
 drivers/mtd/nand/gpmi-nand/gpmi-nand.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
index 9faffd8..b784b39 100644
--- a/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/gpmi-nand/gpmi-nand.c
@@ -2307,8 +2307,8 @@ static const struct of_device_id gpmi_nand_id_table[] = {
 		.data = (void *)&gpmi_devdata_imx6ul,
 	}, {
 		.compatible = "fsl,imx7d-gpmi-nand",
-		.data = &gpmi_devdata_imx7d,
-	}
+		.data = (void *)&gpmi_devdata_imx7d,
+	}, { /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, gpmi_nand_id_table);
 
-- 
1.9.1

